devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/examples/logging-json/java/src/main/java/org/acme/logging/json/LoggingJsonResource.java

```diff
@@ -0,0 +1,53 @@
+package org.acme.logging.json;
+
+import org.jboss.logging.Logger;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.ServerErrorException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+@Path("/logging-json")
+public class LoggingJsonResource {
+
+    private static final Logger LOG = Logger.getLogger(LoggingJsonResource.class);
+
+    private final AtomicInteger speed = new AtomicInteger(5);
+
+    @GET
+    @Path("slowdown")
+    @Produces(MediaType.TEXT_PLAIN)
+    public int slow() {
+        return speed.updateAndGet(i -> {
+            if (i == 0) {
+                LOG.infof("Stop being lazy , you can't go slower than %s", i);
+                return i;
+            }
+            LOG.infof("Come on /faster 🚀, your speed is %s", i);
+            return i - 1;
+        });
+    }
+
+    @GET
+    @Path("faster")
+    @Produces(MediaType.TEXT_PLAIN)
+    public int faster() {
+        final AtomicReference<RuntimeException> e = new AtomicReference<>();
+        int newSpeed = speed.updateAndGet(i -> {
+            if (i == 9) {
+                e.set(new ServerErrorException("You just fell down 🤕, speed is back to " + i, Response.Status.SERVICE_UNAVAILABLE));
```

Comment:
I don't understand this - I fell when I went to speed 9 but writes speed is back to 9 but then set it to 5...that does not feel right.

Reply:
I will just keep one endpoint to go `faster`, from 0 to 10, at eleven you fall and back to 0?